keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertex.java

```diff
@@ -0,0 +1,48 @@
+package io.improbable.keanu.vertices.dbl.probabilistic;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+
+public class HalfGaussianVertex extends GaussianVertex {
+
+    private static final double LOG_TWO = Math.log(2);
+
+    /**
+     * One sigma that match a proposed tensor shape of Gaussian
+     * <p>
+     * If provided parameter is scalar then the proposed shape determines the shape
+     *
+     * @param tensorShape the desired shape of the tensor in this vertex
+     * @param sigma       the sigma of the Gaussian with either the same tensorShape as specified for this vertex or a scalar
+     */
+    public HalfGaussianVertex(int[] tensorShape, DoubleVertex sigma) {
+        super(tensorShape, 0.0, sigma);
+    }
+
+    public HalfGaussianVertex(DoubleVertex sigma) {
+        super(0.0, sigma);
+    }
+
+    public HalfGaussianVertex(double sigma) {
+        super(0.0, new ConstantDoubleVertex(sigma));
+    }
+
+    public HalfGaussianVertex(int[] tensorShape, double sigma) {
+        super(tensorShape, 0.0, new ConstantDoubleVertex(sigma));
+    }
+
+    @Override
+    public double logProb(DoubleTensor value) {
```

Comment:
Nice work here. Will `HalfGaussian` only ever be used with a zeroed `Mu`, or will people want a `HalfGaussian` starting at `2.5` etc? I'll take your opinion on this as gospel

Reply:
Always zero for consistency with ExponentialVertex (we can always add a constant to it with the same effect as a nonzero location).